packages/builder/builder/src/plugins/tsChecker.ts

```diff
@@ -44,6 +45,11 @@ export const builderPluginTsChecker = (): DefaultBuilderPlugin => {
           return;
         }
 
+        const { experiments } = (
+          api as WebpackBuilderPluginAPI
+        ).getNormalizedConfig();
+        const enableSourceBuild = experiments?.sourceBuild ?? false;
```

Comment:
Normally we do not need to set a default value for a normalized config, this default value should be maintained in the default config.

Reply:
Maybe there is no default when using `builder-rspack-provider`?

Reply:
`const config = api.getNormalizedConfig();` 

With this code, I don't have access to the type definition of experiments.sourceBuild.

Reply:
I think it is bacuase `SharedNormalizedConfig` is missing `experiments` field, we can add it.

Reply:
I'll try to add it, you can review it for me.

Reply:
I found that the `sourceBuild` field cannot be added on the `SharedExperimentsConfig` type, as it is currently only available in the webpack builder.

Reply:
So I think it's still necessary to determine if sourceBuild exists and provide a default value.

Reply:
Ok, the type can be improved after we support `sourceBuild` in Rspack provider.